modules/admin-ui/src/main/java/org/opencastproject/adminui/api/MediaPackageLockService.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+package org.opencastproject.adminui.api;
```

Comment:
Hi @JamesUoM and @ts23 , Did you ever consider having the mediapackge lock as an Opencast service versus an Admin UI service? If so, why did you feel that it would be better for this service to be exclusively for use in the Admin UI?

Reply:
I think @ts23 just implemented this as a soln to the problem of combative editing. But you are right, making this a pure "mediapackage" service rather than an "event/mediapackage" would be better. Where should the REST API lie though as at the moment it extends admin-ng/event? I can only think that it should be added to assets?

Reply:
... for us the location may be due to us calling the lock from our "edit" node ie not admin and we are forwarding all event API requests to admin, as that is all the "edit" nodes needs.

Reply:
@JamesUoM Thank you for the responses. I would only suggest refactoring as much as it makes sense to identify a path for iteration 2 of the mediapackage lock service. Since you are most knowledgable about this service, you are best positioned to drop some vision how it could be extended to be used in more use case. For example, concurrent metadata editing or workflow start race conditions on same mp.

Event if mediapackage lock service was it's own module, the admin UI can offer a REST api for it, as an adapter, like it does for other services. The mp lock service could also offer a REST api, or not, just its OSGI wiring.

Reply:
@karendolan and @JamesUoM 

Please be careful to not introduce to notion of locking in an ad-hoc way as locking can be quite problematic. Also, a generic media package lock service would likely have other requirements than the locking for the edit node does. Think of things as recursive locks, deadlock detection and avoidance, etc. which all are quite unproblematic for what James does here but more problematic if used in the backend.